### PR TITLE
Implement SimpleTimeCache

### LINKED ISSUE #168: TimeCache performance issue
**Motivation**
`TimeCache.put()` takes almost 3% of the total time of lodestar application, we need to consider implementing our own cache instead.
Screen Shot 2021-11-30 at 13 08 32

[1130_lodestar_subscribeAllSubnets_local.cpuprofile.zip](https://github.com/ChainSafe/js-libp2p-gossipsub/files/7623334/1130_lodestar_subscribeAllSubnets_local.cpuprofile.zip)

### PR DESCRIPTION
**Pull request by tuyennhv**
**Motivation**
+ Need to implement a faster time cache

**Description**
+ Implement SimpleTimeCache
+ The benchmark test shows that it's 4x faster than the current npm TimeCache

Close #168

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,6 @@
 .idea/
 node_modules/
 src/
-package-lock.json
```
Comment (dapplion) [inline .gitignore:4]: Don't commit `package-lock.json` only `yarn.lock`

File: ts/utils/time-cache.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    this.entries.set(key, Date.now())
+    this.prune()
+  }
+
+  prune (): void {
+    const now = Date.now()
+    if (now - this.lastPruneTime < 200) {
+      return
+    }
+    this.lastPruneTime = now
```
Comment (dapplion) [inline ts/utils/time-cache.ts:34]: So much simpler and elegant :+1: :tada:

File: .gitignore (hunk 1 above)
Comment (tuyennhv) [reply .gitignore:4]: @wemeetagain @dapplion we use npm in all of our github workflows, and we instruct to use it in README as well. Do we really want to switch to yarn?

Also I have to use node 14 and freeze dependencies to get through CI so I have to commit `package-lock.json`

Comment (dapplion) [reply .gitignore:4]: @wemeetagain Can you take a look?

Comment (dapplion) [reply .gitignore:4]: @tuyennhv please stick to yarn.lock in this PR and open an issue about switching to package-lock.json. This PR can't be +14000 lines and we can discuss this latter

Comment (tuyennhv) [reply .gitignore]: @dapplion I just rebase to master, we don't need to commit `package-lock.json` anymore

**Review by dapplion — APPROVED** [review]

**Pull request merged**